_posts/2020-12-18-react-generic-context.md

```diff
@@ -0,0 +1,232 @@
+---
+layout: post
+authors: [orjan_de_smet]
+title: "Making a generic Context in React"
+image: /img/react-generic-context/banner.jpg
+tags: [React]
+category: Architecture
+comments: true
+---
+
+## A background
+<img class="image fit center" alt="A picture of the Windows XP background" src="/img/react-generic-context/a-background.jpg">
+
+I have been mostly working with Angular for the past 4 years.
+I am, however, a firm believer that no one framework or library is better than all others and I love a sporadic challenging dive into the unknown.
+
+Those who know a little bit about React will know that, like Angular, Vue and others, it’s a JavaScript library to create web applications by means of components.
+These components are considered building blocks and can consist themselves of smaller components.
+These components have properties, being passed by their parents and can these properties can influence the rendered view and logic of the components.
+
+Those with a bit more React experience, will know that in a larger tree of components, it’s a nuisance to get a property from a component near the root of the tree to a component near the leafs of the tree by simple property binding.
+It also muddies intermediary components with irrelevant code.
+
+Fortunately React includes **Context** [Docs](https://reactjs.org/docs/context.html){:target="_blank" rel="noopener noreferrer"}.
+It provides a way to share value like preferences/themes and authentication data without having to explicitly set those values in each component’s properties.
+
+The developer basically creates a `Context` variable which is provided by a near root component using a `Provider`, and can be consumed by other components  using either a `Consumer` or a `useContext` hook.
```

Comment:
* What is a root component? Do you mean container component? 

* using a `Provider` component
* eithar a `Consumer` component or a `useContext` hook